### ISSUE TITLE
Infinite loop with Layout/FirstArgumentIndentation and Layout/LineLength causes massive whitespace corruption

### ISSUE THREAD

**Issue by maxamus123** [rubocop/rubocop#14817]
## Expected behavior

Running `rubocop --autocorrect` should fix code style issues without corrupting the file.

## Actual behavior

When auto-correcting the following code, RuboCop enters an infinite loop between `Layout/FirstArgumentIndentation` and `Layout/LineLength`, continuously adding whitespace until lines become thousands of characters long.

The error message:
```
Infinite loop detected in /Users/maxwellcreamer/projects/homeshool/homeschool-api-main/app/controllers/api/v1/course_sequences_controller.rb and caused by Layout/FirstArgumentIndentation, Layout/LineLength
```

The resulting file had lines with thousands of spaces, e.g.:
```ruby
# Before autocorrect (clean code):
sequences = CourseSequence
  .includes(
    course_sequence_memberships: {
      course: :course_components
    }
  )
  .joins(:courses)

# After infinite loop (corrupted with thousands of spaces):
sequences =                                                    [... thousands of spaces ...]                                                                                                                                                                                                                       CourseSequence
                                                                                                                                                                                                                                                                          .includes(
# etc.
```

## Steps to reproduce the problem

1. Create file with this code:

```ruby
# frozen_string_literal: true

module Api
  module V1
    class CourseSequencesController < ApplicationController
      before_action :authenticate_user!

      def index
        sequences = build_sequences_query
        render json: { course_sequences: sequences }
      end

      private

      def build_sequences_query
        sequences = CourseSequence
          .includes(
            course_sequence_memberships: {
              course: :course_components
            }
          )
          .joins(:courses)

        sequences.distinct.order(id: :asc)
      end
    end
  end
end
```

2. Use this `.rubocop.yml`:

```yaml
AllCops:
  TargetRubyVersion: 3.2
  NewCops: enable

Layout/FirstArgumentIndentation:
  EnforcedStyle: consistent

Layout/ArgumentAlignment:
  EnforcedStyle: with_fixed_indentation

Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

3. Run `rubocop --autocorrect` on the file

## RuboCop version

```
1.84.0 (using Parser 3.3.10.1, rubocop-ast 1.49.0, analyzing as Ruby 3.2, running on ruby 3.2.6) [arm64-darwin24]
```

## Notes

This appears to be a regression or variant of #12261 which was supposed to be fixed. The combination of `Layout/FirstArgumentIndentation` with `EnforcedStyle: consistent` and `Layout/MultilineMethodCallIndentation` with `EnforcedStyle: indented` seems to create a conflict where one cop keeps adding indentation and the other keeps trying to adjust it, leading to runaway whitespace addition.
